src/Computation/Values/Symbolics/Write.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Numerics;
+using Microsoft.Z3;
+using Symbolica.Computation.Values.Constants;
+using Symbolica.Expression;
+
+namespace Symbolica.Computation.Values.Symbolics;
+
+internal sealed class Write : BitVector, IWriteValue
+{
+    private readonly IValue _writeBuffer;
+    private readonly IValue _writeOffset;
+    private readonly IValue _writeValue;
+
+    public Write(IValue writeBuffer, IValue writeOffset, IValue writeValue)
+        : base(writeBuffer.Size)
+    {
+        _writeBuffer = writeBuffer;
+        _writeOffset = writeOffset;
+        _writeValue = writeValue;
+    }
+
+    public override BitVecExpr AsBitVector(Context context)
+    {
+        return Flatten().AsBitVector(context);
+    }
+
+    public IValue Read(IValue offset, Bits size)
+    {
+        var readMask = Mask(offset, size);
+        var writeMask = Mask(_writeOffset, _writeValue.Size);
+
+        return And(readMask, writeMask) is IConstantValue a && a.AsUnsigned().IsZero
+            ? _writeBuffer is IWriteValue b
+                ? b.Read(offset, size)
+                : new Read(_writeBuffer, offset, size)
+            : Xor(readMask, writeMask) is IConstantValue x && x.AsUnsigned().IsZero
+                ? _writeValue
+                : new Read(Flatten(), offset, size);
+    }
+
+    private IValue Flatten()
+    {
+        var writeMask = Mask(_writeOffset, _writeValue.Size);
+        var writeData = ShiftLeft(ZeroExtend(Size, _writeValue), _writeOffset);
+
+        return Or(And(_writeBuffer, Not(writeMask)), writeData);
+    }
+
+    private IValue Mask(IValue offset, Bits size)
+    {
+        return ShiftLeft(ZeroExtend(Size, Not(ConstantUnsigned.Create(size, BigInteger.Zero))), offset);
+    }
+
+    private static IValue And(IValue left, IValue right)
```

Comment:
These functions seem generally useful, should we move them to a static class (I think you lot call it a factory 😉) so they can be used throughout these Value classes? The other option would be to have static factory methods on `And` etc and hide the constructors, but then having them in the `Symbolics` namespace would be possibly incorrect.

Reply:
I was planning to maybe do the latter of these options, but I want to see what it all looks like once the AND\OR changes are in and the two `IExpression` implementations have hopefully been rolled into one. So we'll leave it for now since they are only used here.